packages/core/src/components/AdaptiveGrid/index.tsx

```diff
@@ -0,0 +1,99 @@
+import React from 'react';
+import withStyles, { WithStylesProps } from '../../composers/withStyles';
+
+type State = {
+  items: number;
+};
+
+export type AdaptiveGridProps = {
+  /** List of width/item pairs, describes how many items to display for windows of at least that width. */
+  breakpoints?: { [key: number]: number };
+  /** Items per row for screens smaller than the smallest breakpoint. */
+  defaultItemsPerRow?: number;
+  /** Removes padding between items. */
+  noGutter?: boolean;
+};
+
+type Props = AdaptiveGridProps & WithStylesProps;
+
+class AdaptiveGrid extends React.PureComponent<Props, State> {
+  static defaultProps = {
+    breakpoints: {},
+    defaultItemsPerRow: 1,
+    noGutter: false,
+  };
+
+  state = {
+    items: this.getItems(),
+  };
+
+  componentDidMount() {
+    window.addEventListener('resize', this.handleResize);
```

Comment:
i wish we could solve this without listening for the resize event :/ can you explore taking the breakpoints and setting the styles based on that? css grid might help here - per media query, setting the css grid column count or something. `cx` can handle media queries, right @milesj ?

Reply:
Yeah it can. Supports any CSS.